src/api/app/models/maintenance_incident.rb

```diff
@@ -7,70 +7,86 @@ class MaintenanceIncident < ActiveRecord::Base
     return if self.released_at
 
-    # set current time, to be used 
+
+    # set current time, to be used
     self.released_at = Time.now.utc
     self.name = patch_name
 
+    counterType = ""
+    values = []
+
     # Run an atomar counter++ based on the used scheme
     if template =~ /%Y/
-      counterType = " AND year  = " + self.released_at.year.to_s
-      year = "'" + self.released_at.year.to_s + "'"
+      counterType << " AND year = ? "
+      values << self.released_at.year.to_s
+      year =  self.released_at.year.to_s
     else
-      counterType = " AND ISNULL(year)"
+      counterType = " AND ISNULL(year) "
       year = "NULL"
     end
     if template =~ /%M/
-      counterType << " AND month = " + self.released_at.month.to_s
-      month = "'" + self.released_at.month.to_s + "'"
+      counterType << " AND month = ? "
+      values << self.released_at.month.to_s
+      month = self.released_at.month.to_s
     else
-      counterType << " AND ISNULL(month)"
+      counterType << " AND ISNULL(month) "
       month = "NULL"
     end
+
     if template =~ /%D/
-      counterType << " AND day   = " + self.released_at.day.to_s
-      day = "'" + self.released_at.day.to_s + "'"
+      counterType << " AND day   = ? "
+      values << self.released_at.day.to_s
+      day = self.released_at.day.to_s
     else
-      counterType << " AND ISNULL(day)"
+      counterType << " AND ISNULL(day) "
       day = "NULL"
     end
     if template =~ /%N/
-      name = ActiveRecord::Base.connection.quote(self.name||"")
-      counterType << " AND name   = " + name
+      name = (self.name||"")
+      counterType << " AND name   = ? " 
+      values << name
     else
-      counterType << " AND ISNULL(name)"
+      counterType << " AND ISNULL(name) "
       name = "NULL"
     end
-    sql = ActiveRecord::Base.connection()
-    r = sql.execute( "SELECT counter FROM updateinfo_counter WHERE maintenance_db_project_id = " + self.maintenance_db_project.id.to_s + counterType + " FOR UPDATE" ).first
+    
+    r = MaintenanceIncident.exec_query(["SELECT counter FROM updateinfo_counter WHERE maintenance_db_project_id = ? #{counterType} FOR UPDATE",
+                                        self.maintenance_db_project.id,counterType,*values]).first
+
     unless r
       # no counter exists, initialize it and select again
-      sql.execute( "INSERT INTO updateinfo_counter(maintenance_db_project_id, year, month, day, name) VALUES('" + self.maintenance_db_project.id.to_s + "', " + year + ", " + month + ", " + day + "," + name + ")" )
-      r = sql.execute( "SELECT counter FROM updateinfo_counter WHERE maintenance_db_project_id = " + self.maintenance_db_project.id.to_s + counterType + " FOR UPDATE" ).first
+      r = MaintenanceIncident.exec_query ["INSERT INTO updateinfo_counter(maintenance_db_project_id, year, month, day, name) VALUES(?,?,?,?,?)",
+                                          self.maintenance_db_project.id,year,month,day,name]
+
```

Comment:
no, "NULL" for month and day. It's a difference if you put "NULL" in a constructed SQL query or in a bind variable